kolibri/core/public/api.py

```diff
@@ -146,3 +156,77 @@ def get_public_file_checksums(request, version):
         json.dumps({"id": error_constants.NOT_FOUND, "metadata": {"view": ""}}),
         content_type="application/json",
     )
+
+
+class SyncQueueViewSet(viewsets.ViewSet):
+    def list(self, request):
+        """Returns length of the queue for each of the available facilities"""
+        SyncQueue.clean_stale()  # first, ensure not expired devices are in the queue
+        facilities = Facility.objects.all()
+        queue = {}
+        for facility in facilities:
+            queue[facility.id] = SyncQueue.objects.filter(
+                user__facility=facility
+            ).count()
+        return Response(queue)
+
+    def check_queue(self):
+        current_transfers = TransferSession.objects.filter(active=True).count()
+        if current_transfers <= MAX_CONCURRENT_SYNCS:
+            allow_sync = True
+            data = {"action": SYNC}
+        else:
+            polling = current_transfers + SyncQueue.objects.all().count()
+            data = {
+                "action": QUEUED,
+                "keep_alive": polling,
+            }
+            allow_sync = False
+        return (allow_sync, data)
+
+    def create(self, request):
+        SyncQueue.clean_stale()  # first, ensure not expired devices are in the queue
+        device_info = get_device_info()
+        if device_info["subset_of_users_device"]:
+            content = {"I'm a Subset of users device": "Nothing to do here"}
+            # would love to use HTTP 418, but it's not fully usable in browsers
+            return Response(content, status=status.HTTP_404_NOT_FOUND)
```

Comment:
Could we maybe do something other than 404 here, just to indicate that this was found, but this device is not to be synced with?

Reply:
Replaced by a 400